### ISSUE TITLE
Path(...).open() does not trigger `consider-using-with`

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#6906]
`Path(...).open()` does not trigger `consider-using-with`  (bug)
`open(...)` does. (good)

**Comment by socketpair** [comment]
Please note, this should also work:

```python
class MyPath(PosixPath):
   ....
```

`MyPath(....).open(...)` should also trigger this.

**Comment by mbyrnepr2** [comment]
Closing since [this is now available](https://github.com/PyCQA/pylint/issues/7964) thanks to the great work by @GideonBear!

Here are some snippets of testcases from that work if you'd like to quickly check it out:
```python
from pathlib import Path, PosixPath


with Path("foo").open(encoding="utf8") as file:  # must not trigge
    _ = file.read()

_ = Path("foo").open(encoding="utf8")

class MyPath(PosixPath):
   ...
_ = MyPath("foo").open(encoding="utf8")
```
